packages/fxa-auth-server/lib/routes/oauth/index.js

```diff
@@ -278,39 +281,43 @@ module.exports = (log, config, oauthdb, db, mailer, devices) => {
         // done with 'session_token_id' at this point, do not return it.
         delete grant.session_token_id;
 
-        // attempt to record metrics, but swallow the error if one is thrown.
-        try {
-          let uid = sessionToken && sessionToken.uid;
+        if (scopeSet.contains(OAUTH_SCOPE_OLD_SYNC)) {
+          // attempt to record metrics, but swallow the error if one is thrown.
+          try {
+            let uid = sessionToken && sessionToken.uid;
 
-          // As mentioned in lib/routes/utils/oauth.js, some grant flows won't
-          // have the uid in `credentials`, so we get it from the oauth DB.
-          if (!uid) {
-            const tokenVerify = await oauthdb.checkAccessToken(
-              grant.access_token
-            );
-            uid = tokenVerify.user;
-          }
+            // As mentioned in lib/routes/utils/oauth.js, some grant flows won't
+            // have the uid in `credentials`, so we get it from the oauth DB.
+            if (!uid) {
+              const tokenVerify = await oauthdb.checkAccessToken(
+                grant.access_token
+              );
+              uid = tokenVerify.user;
+            }
 
-          const account = await db.account(uid);
-          const ecosystemAnonId = account.ecosystemAnonId;
+            const account = await db.account(uid);
+            const ecosystemAnonId = account.ecosystemAnonId;
 
-          await request.emitMetricsEvent('oauth.token.created', {
-            grantType: request.payload.grant_type,
-            uid,
-            ecosystemAnonId,
-            clientId: request.payload.client_id,
-          });
+            await request.emitMetricsEvent('oauth.token.created', {
```

Comment:
I think it's important that we continue to emit this event even when not using the `oldsync` scope; would it work here if we reduce the `if contains SCOPE_OLD_SYNC` condition to *just* cover the emission of the `account.signed` event? (This would mean that the `oauth.token.created` event doesn't get `service: 'sync'`, but I don't think we've ever had that data for that event in the past).